assets/src/design-system/components/switch/index.js

```diff
@@ -0,0 +1,233 @@
+  min-width: unset !important;
+  appearance: none;
+  border: 0 !important;
+  box-shadow: none !important;
+  margin: -1px !important;
+  outline: none !important;
+  overflow: hidden;
+`;
+
+const RadioButtonLabel = styled(Text).attrs({ forwardedAs: 'label' })`
+  box-sizing: border-box;
+  flex: 1;
+  width: 50%;
+  padding: 0px 6px;
+  z-index: 1;
+  color: ${({ isActive, theme }) =>
+    isActive ? theme.colors.inverted.fg.primary : theme.colors.fg.secondary};
+  cursor: pointer;
+  overflow: hidden;
+  text-align: center;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  transition: color 0.15s ease-out;
+
+  ${({ disabled }) =>
+    disabled &&
+    css`
+      cursor: default;
+      opacity: 0.3;
+    `}
+
+  /* add focus styling on the slider when the hidden input is focused */
+  :focus-within ~ span {
+    ${themeHelpers.focusCSS};
+  }
+`;
+
+const SlidingButton = styled.span`
+  box-sizing: border-box;
+  position: absolute;
+  display: block;
+  top: 0;
+  left: 0;
+  width: 50%;
+  height: ${SWITCH_HEIGHT}px;
+  border-radius: 100px;
+  background-color: ${({ theme }) => theme.colors.interactiveBg.primaryNormal};
+  transition: left 0.15s ease-out;
+  z-index: 0;
+
+  ${({ hasOffset }) => hasOffset && `left: 50%`}
+`;
+
+/**
+ * A controlled radio group that looks like a switch.
+ *
+ * Keyboard navigation:
+ * - [Tab] can be used to navigate between radio groups
+ * - [Enter], [Space], [ArrowLeft], [ArrowRight] [ArrowUp], and [ArrowDown] can be used to select radio buttons
+ *
+ * @param {Object} props The props
+ * @param {string} props.className className that is applied to the outer container
+ * @param {string} props.disabled whether the radio buttons are disabled
+ * @param {string} props.groupLabel the label for the radio group. Used for accessibility
+ * @param {string} props.id unique id used for accessibility
+ * @param {string} props.name name of the radio group
+ * @param {string} props.offLabel label for the 'off' radio button
+ * @param {Function} props.onChange change event handler
+ * @param {string} props.onLabel label for the 'on' radio button
+ * @param {boolean} props.value the value of the radio group
+ * @return {Object} The radio button
+ */
+export const Switch = forwardRef(function (
+  {
+    className = '',
+    disabled,
+    groupLabel,
+    id,
+    offLabel,
+    onChange,
+    onLabel,
+    value,
+    ...props
+  },
+  ref
+) {
+  const radioGroupRef = useRef(ref);
+  const ids = useMemo(
+    () => ({
+      group: id || uuidv4(),
+      offInput: uuidv4(),
+      onInput: uuidv4(),
+    }),
+    [id]
+  );
+
+  const handleChange = useCallback(
+    (evt) => {
+      onChange(evt, evt.target.value === 'true');
+    },
+    [onChange]
+  );
+
+  const handleKeyDown = useCallback(
+    (evt) => {
+      onChange(evt, !value);
+    },
+    [onChange, value]
+  );
+
+  useKeyDownEffect(radioGroupRef, ['space', 'enter'], handleKeyDown, [
+    handleKeyDown,
+    value,
+  ]);
```

Comment:
The switch gets the accessibility from being a radio group. Maybe we shouldn't add this here and instead add it where we need it (in the editor) since it adds extra behavior that may not be wanted by all consumers of this component.

Reply:
i like having the useKeyDownEffect here if we're expecting switches to have the same behavior across the board. maybe i am reading this comment wrong.

Reply:
I think I'm leaning into removing it tbh (and then adding it back in in the editor). This keydown effect overrides normal browser functionality.

If this component were ever used in the future but they didn't want `space` and `enter` to change the value, they'd have to come into this component and change it first rather than the base component displaying the 'normal' browser interactions